storage/src/main/java/tech/pegasys/teku/storage/server/pruner/BlockPruner.java

```diff
@@ -100,8 +103,16 @@ private void pruneBlocks() {
     }
     LOG.info("Pruning finalized blocks before slot {}", earliestSlotToKeep);
     try {
-      database.pruneFinalizedBlocks(earliestSlotToKeep.decrement());
-      LOG.info("Finalized blocks before slot {} have been pruned.", earliestSlotToKeep);
+      final boolean allPruned =
+          database.pruneFinalizedBlocks(earliestSlotToKeep.decrement(), pruneLimit);
+      if (allPruned) {
+        LOG.info("Finalized blocks before slot {} have been pruned.", earliestSlotToKeep);
+      } else {
+        LOG.info(
+            "{} finalized blocks before slot {} have been pruned, pruning not finished.",
+            pruneLimit,
+            earliestSlotToKeep);
+      }
```

Comment:
i wonder if a better way to lay this out, is pruneFinalizedBlocks changes to return a UInt64 of the last block deleted...
Basically then we can say something like
'Blocks up to slot N have been deleted', and compare the `earliestSlotToKeep.decrement()` value to the last block deleted to determine if all blocks got deleted (actually thats less easy because empty slots, but if its within an epoch i'd say all were deleted)

thoughts?

Reply:
@rolfyone how about this? https://github.com/Consensys/teku/pull/7483/commits/7a41c3a9b6fbbd248dd8140b57764e76b3661521
Even simpler than your suggestion plus we don't really care much how correct is the return value. The only payback is 1 log.info line correctness once in the ages

Reply:
that works